marshal.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package httprequest
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+
+	"github.com/julienschmidt/httprouter"
+	"gopkg.in/errgo.v1"
+)
+
+// Marshal takes the input structure and creates an http request.
+//
+// See: Unmarshal for more details.
+//
+// For fields with a "path" item in the structural tag, the request uri must
+// contain a placeholder with its name.
+// Example:
+// For
+//    type Test struct {
+//	    username string `httprequest:"user,path"`
+//    }
+// ...the request url must contain a "::user::" placeholder:
+//    http://localhost:8081/:user/files
```

Comment:
How else would you know how to marshal a field to the request URL? Anyway: this mimics the httprouter.